### PR TITLE
Bug 2068181: Fix Event source powered with kamelet type source to show associated resources in the sidepanel

### PR DESCRIPTION
**Pull request by debsmita1**
**Fixes:**
https://issues.redhat.com/browse/OCPBUGSM-42451

**Root analysis:**
Was fetching the `kameletType` from the wrong property 

**Screenshot:**
Screenshot from 2022-04-08 21-08-16

**Test coverage:**
test

/kind bug

### CODE REVIEW

**Comment by invincibleJai** [comment]
Does any of these kamelet sources/evnet source doesn't have associated deployments? if it's there then should be shown in the resources tab. So we need to test with one such resource kamelet type telegram should be having

**Comment by sahil143** [comment]
I am only seeing Target and Pods No Deployment. Am I missing something?
image

Pods Section is loading with a bit of delay?
Kapture 2022-04-11 at 20 50 02

**Comment by debsmita1** [comment]
> I am only seeing Target and Pods No Deployment. Am I missing something? image

Deployment is loading for me

> 
> Pods Section is loading with a bit of delay? Kapture 2022-04-11 at 20 50 02

Yes happening for other nodes also where we show pods in the resources tab

File: frontend/packages/knative-plugin/src/components/overview/EventSourceResources.tsx [hunk 1]
```diff
@@ -1,46 +1,42 @@
...
   obj: K8sResourceKind;
-  ownedSources?: K8sResourceKind[];
 };
 
-const EventSourceResources: React.FC<EventSourceResourcesProps> = ({ obj, ownedSources }) => {
+type EventSourceDeploymentsProps = {
+  deploymentObj: K8sResourceKind;
+};
+
+type ventSourceOwnedEventSourcesProps = {
```
Comment (invincibleJai) [inline frontend/packages/knative-plugin/src/components/overview/EventSourceResources.tsx:24]:
```suggestion
type evntSourceOwnedEventSourcesProps = {
```

File: frontend/packages/knative-plugin/src/components/overview/EventSourceResources.tsx [hunk 2]
```diff
@@ -70,27 +66,42 @@ const EventSourceResources: React.FC<EventSourceResourcesProps> = ({ obj, ownedS
...
-        ownedSources.map((source) => (
-          <EventSourceOwnedList key={source.metadata.uid} source={source} />
-        ))}
+      ) : null}
     </>
   );
 };
 
-export default EventSourceResources;
+export const EventSourceOwnedEventSources: React.FC<ventSourceOwnedEventSourcesProps> = ({
```
Comment (invincibleJai) [inline frontend/packages/knative-plugin/src/components/overview/EventSourceResources.tsx:97]:
```suggestion
export const EventSourceOwnedEventSources: React.FC<evntSourceOwnedEventSourcesProps> = ({
```

and this name seems confusig `EventSourceOwnedEventSources`

Comment (debsmita1) [reply frontend/packages/knative-plugin/src/components/overview/EventSourceResources.tsx]: Change it to `OwnedEventSources`

File: frontend/packages/knative-plugin/src/topology/sidebar/knative-eventsource-tab-sections.tsx [hunk 3]
```diff
@@ -18,14 +22,52 @@ export const getKnativeSidepanelSinkSection = (element: GraphElement) => {
...
+  }
+  return undefined;
+};
+
+export const getKnativeSidepanelSinkEventSources = (element: GraphElement) => {
+  const resource = getResource(element);
+  const data = element.getData();
+  if (!resource) {
+    return undefined;
+  }
```
Comment (invincibleJai) [inline frontend/packages/knative-plugin/src/topology/sidebar/knative-eventsource-tab-sections.tsx:62]: we can do early return if resources also don't exists and any reason to return `undefined` over `null` ?

```suggestion
  if (!resource || !data?.resources) {
    return null;
  }
```

File: frontend/packages/knative-plugin/src/topology/sidebar/knative-eventsource-tab-sections.tsx [hunk 4]
```diff
@@ -18,14 +22,52 @@ export const getKnativeSidepanelSinkSection = (element: GraphElement) => {
...
+  if (
+    isDynamicEventResourceKind(referenceFor(resource)) ||
+    (resource.kind === CamelKameletBindingModel.kind &&
+      data.data.kameletType === KameletType.Source)
+  ) {
+    return (
+      <TopologySideBarTabSection>
+        <OwnedEventSources eventSources={element.getData().resources?.eventSources} />
```
Comment (invincibleJai) [inline frontend/packages/knative-plugin/src/topology/sidebar/knative-eventsource-tab-sections.tsx:70]:
```suggestion
        <OwnedEventSources eventSources={data.resources.eventSources} />
```

File: frontend/packages/knative-plugin/src/topology/sidebar/knative-eventsource-tab-sections.tsx [hunk 5]
```diff
@@ -18,14 +22,52 @@ export const getKnativeSidepanelSinkSection = (element: GraphElement) => {
...
+    return (
+      <TopologySideBarTabSection>
+        <EventSourceDeployments deploymentObj={element.getData().resources?.associatedDeployment} />
+      </TopologySideBarTabSection>
+    );
+  }
+  return undefined;
+};
+
+export const getKnativeSidepanelSinkEventSources = (element: GraphElement) => {
```
Comment (invincibleJai) [inline frontend/packages/knative-plugin/src/topology/sidebar/knative-eventsource-tab-sections.tsx:57]: @debsmita1 can we add test for such components as well?

File: frontend/packages/knative-plugin/src/topology/sidebar/knative-eventsource-tab-sections.tsx [hunk 6]
```diff
@@ -18,14 +22,52 @@ export const getKnativeSidepanelSinkSection = (element: GraphElement) => {
...
+  if (
+    isDynamicEventResourceKind(referenceFor(resource)) ||
+    (resource.kind === CamelKameletBindingModel.kind &&
+      data.data.kameletType === KameletType.Source)
+  ) {
+    return (
+      <TopologySideBarTabSection>
+        <EventSourceDeployments deploymentObj={element.getData().resources?.associatedDeployment} />
+      </TopologySideBarTabSection>
+    );
```
Comment (invincibleJai) [inline frontend/packages/knative-plugin/src/topology/sidebar/knative-eventsource-tab-sections.tsx:52]:
```suggestion
    return (
      <TopologySideBarTabSection>
        <EventSourceDeployments deploymentObj={data.resources.associatedDeployment} />
      </TopologySideBarTabSection>
    );
```

File: frontend/packages/knative-plugin/src/topology/sidebar/knative-eventsource-tab-sections.tsx [hunk 7]
```diff
@@ -18,14 +22,52 @@ export const getKnativeSidepanelSinkSection = (element: GraphElement) => {
...
+  }
+  return undefined;
+};
+
+export const getKnativeSidepanelSinkAssociatedDeployment = (element: GraphElement) => {
+  const resource = getResource(element);
+  const data = element.getData();
+  if (!resource) {
+    return undefined;
+  }
```
Comment (invincibleJai) [inline frontend/packages/knative-plugin/src/topology/sidebar/knative-eventsource-tab-sections.tsx:42]:
```suggestion
  if (!resource || !data?.resources) {
    return null;
  }
```

File: frontend/packages/knative-plugin/src/topology/sidebar/knative-eventsource-tab-sections.tsx (hunk 5 above)
Comment (debsmita1) [reply frontend/packages/knative-plugin/src/topology/sidebar/knative-eventsource-tab-sections.tsx:60]: added tests

**Comment by invincibleJai** [comment]
@debsmita1 for the empty state of pods the message doesn't seem right here and atm no pods exist so shouldn't even show this section as telegram source create deployment not KSVC. 

image

and on event sink creation flow , i don't see resources though have created channel

image

**Comment by debsmita1** [comment]
> @debsmita1 for the empty state of pods the message doesn't seem right here and atm no pods exist so shouldn't even show this section as telegram source create deployment not KSVC.
> 
> image
> 
Using the default empty text if there are no pods , and also showing the `Source` heading when there are no sources
Screenshot from 2022-04-20 23-10-09


> and on event sink creation flow , i don't see resources though have created channel
> 
> image

couldn't reproduce the issue
c-es

**Comment by debsmita1** [comment]
> @debsmita1 for the empty state of pods the message doesn't seem right here and atm no pods exist so shouldn't even show this section as telegram source create deployment not KSVC.
> 
> image
> 
Using the default empty text if there are no pods , and also showing the `Source` heading when there are no sources
Screenshot from 2022-04-20 23-10-09


> and on event sink creation flow , i don't see resources though have created channel
> 
> image

couldn't reproduce the issue
c-es

**Pull request merged**